### PR TITLE
chore: don't generate theia resources; remove sidecars; add devfile.yaml to work with the registry

### PR DESCRIPTION
**Pull request by svor**
### What does this PR do?

- Don't generate devworkspaces with che-theia editor
- Remove dockerfiles that used to use in dev components
- Add devfile.yaml to develop che-devfile-registry on the dogfooding cluster 

### Screenshot/screencast of this PR

I've used a devfile.yaml provided by the current changes to test if the registry works. For this I prepared OS instance with Che, added build capabilities to the cluster and:
1. Build registry by executing **1. Build and Publish a Custom Devfile Registry** command: 
screenshot-nimbusweb me-2023 03 30-15_52_46

2. Executed a command to use resulted image: **2. Configure Che to use the Custom Devfile Registry**
screenshot-nimbusweb me-2023 03 30-15_56_28

3. Checked che-devfile-registry pod:
screenshot-nimbusweb me-2023 03 30-15_58_04

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/22091

### How to test this PR?

- Build che-devfile registry
- Check generated `htdocs/devfiles/stack/` folders, they should contain generated devworkspaces only for che-code and che-idea editors
 
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: patch-checluster.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+#
+# Copyright (c) 2018-2023 Red Hat, Inc.
```
Comment (nickboldt) [inline patch-checluster.sh:3]: technically, this started in 2023. :D

File: devfile.yaml [hunk 2]
```diff
@@ -1,72 +1,39 @@
...
+          set -o xtrace;
+          rm -rf output;
+          mkdir output;
+          BUILDER=podman SKIP_TEST=true SKIP_FORMAT=true SKIP_LINT=true NODE_BUILD_OPTIONS="-max-old-space-size=1024" ./build.sh &&
+          export IMAGE=image-registry.openshift-image-registry.svc:5000/$(kubectl get sa default -o json | jq -r '.metadata.namespace')/che-devfile-registry &&
+          podman tag quay.io/eclipse/che-devfile-registry:next ${IMAGE} &&
+          podman login --tls-verify=false -u $(oc whoami | tr -d :) -p $(oc whoami -t) image-registry.openshift-image-registry.svc:5000 &&
+          podman push --tls-verify=false "${IMAGE}"
+      component: builder
+      label: 1. Build and Publish a Custom Devfile Registry
```
Comment (nickboldt) [inline devfile.yaml:29]: for consistency, should the `label` be at the start of each `exec` section?

File: README.md [hunk 3]
```diff
@@ -1,6 +1,31 @@
...
+---
+displayName: Python
+description: Python Stack with Python 3.8
+tags: ["Community", "Centos", "Python", "pip"]
+icon: /images/python.svg
+links:
+  v2: https://github.com/che-samples/python-hello-world/tree/main
+```
+
+Here are all the supported values:
```
Comment (nickboldt) [inline README.md:17]: what follows is a verbatim copy of the sample meta.yaml, not an explanation of the supported values. Copy and paste error?

File: README.md [hunk 4]
```diff
@@ -1,6 +1,31 @@
 # Eclipse Che devfile registry
 
-This repository holds ready-to-use Devfiles for different languages and technologies.
+This repository holds meta.yaml files that refer to the samples with Devfiles.
+
+Here is the expected format of a `meta.yaml`:
```
Comment (nickboldt) [inline README.md:5]:
```suggestion
Here is an example `meta.yaml` file:
```

File: devfile.yaml [hunk 5]
```diff
@@ -1,72 +1,39 @@
...
+  - id: configure-custom-devfile-registry
+    exec:
+      label: "2. Configure Che to use the Custom Devfile Registry"
+      component: builder
+      commandLine: |
+          export IMAGE=image-registry.openshift-image-registry.svc:5000/$(kubectl get sa default -o json | jq -r '.metadata.namespace')/che-devfile-registry &&
+          export CHECLUSTER_NAMESPACE="$(kubectl get checluster --all-namespaces -o json | jq -r '.items[0].metadata.namespace')" &&
+          kubectl create secret -n "${CHECLUSTER_NAMESPACE}" docker-registry regcred --docker-server=image-registry.openshift-image-registry.svc:5000 --docker-username=$(oc whoami | tr -d :) --docker-password=$(oc whoami -t) &&
+          kubectl patch serviceaccount default -n "${CHECLUSTER_NAMESPACE}" -p '{"imagePullSecrets": [{"name": "regcred"}]}' &&
+          ./patch-checluster.sh "${IMAGE}"
```
[the remark this answers is not in the document]
Comment (svor) [reply devfile.yaml:39]: yes, it's how i tested current changes :)

File: README.md (hunk 3 above)
Comment (svor) [reply README.md:17]: added more comments

File: devfile.yaml (hunk 2 above)
Comment (svor) [reply devfile.yaml]: fixed

**Review by nickboldt — APPROVED** [review]

**Pull request merged**